tests/test_black.py

```diff
@@ -1727,6 +1727,16 @@ def test_nested_gitignore(self) -> None:
         )
         self.assertEqual(sorted(expected), sorted(sources))
 
+    def test_invalid_gitignore(self) -> None:
+        path = THIS_DIR / "data" / "invalid_gitignore_tests"
+        empty_config = THIS_DIR / "data" / "invalid_gitignore_tests" / "pyproject.toml"
+        result = BlackRunner().invoke(
+            black.main, ["--verbose", "--config", str(empty_config), str(path)]
+        )
+        assert result.exit_code == 1
+        assert result.stderr_bytes is not None
+        assert b"Invalid git pattern: '!\\n'" in result.stderr_bytes
```

Comment:
This is the error emitted by pathspec 0.9.0 thanks to https://github.com/cpburnz/python-path-specification/pull/50

Reply:
It seems fragile to test the exact error message produced by pathspec. Maybe we should check only for the "Could not parse" bit from black's own code?

Reply:
Makes sense to me! Especially if we'd be upgrading the dependency quietly.